walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,82 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<String> relevantAttendees = new ArrayList<>();
+    request.getAttendees().forEach(attendee -> relevantAttendees.add(attendee));
+
+    List<TimeRange> affectedTimeRanges = new ArrayList<>(); 
+    // get all affected time ranges
+    for (Event event : events) {
+      Set<String> scheduledAttendees = event.getAttendees(); 
+      for (String attendee : relevantAttendees) {
+        if (scheduledAttendees.contains(attendee)) {
+          // this event affects the final result
+          affectedTimeRanges.add(event.getWhen());
+          break;
+        }
+      }
+    }
+
+    long minRequiredDuration = request.getDuration();
+    long maxDuration = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true).duration();
+
+    List<TimeRange> availableRanges = new ArrayList<>();
+    if (minRequiredDuration > maxDuration) {
+      return availableRanges;
```

Comment:
Please add comment explaining why the max event duration is 1 day. And why that there cannot be event longer than 1 day.

E.g, if affectedTimeRanges is empty, and minRequiredDuration is two day, should we still return 0 available ranges? Or is there a requirement on event duration that I'm not aware of?

Reply:
Hmm... But given how `TimeRange` is implemented, I don't think it is possible? Is it? :O This is the only reason. Should I just state that? However, personally, I found it quite obvious since the earliest start time of `TimeRange` is `START_OF_DAY` and the latest end time is `END_OF_DAY`.

Reply:
I see, thanks for clarification. However, I would still suggest to give a comment here, this seems to be a caveat (that max duration is <= 1 day) in the current design that you may want to highlight